README.md

```diff
@@ -44,46 +42,321 @@ see https://github.com/ioBroker/ioBroker/wiki/Installation
+
+### Multihost
+**Feature Status: Stable**
+
+When local interfaces are needed or the ressources are all used the ioBroker system can be extended by additional hosts. They all connect to the same states and objects databases and are so synced with each other.
+
+When the states and/or objects databases are provided by an js-controller process on one host this process need to be configured for these dstabases to bind to port 0.0.0.0 (enter this in ```iobroker setup custom```) instead of 127.0.0.1
+
+All other hosts are configured to connect to this master host.
+
+For detailed setup instructions see https://www.iobroker.net/docu/index-24.htm?page_id=3068&lang=de
+
+
+### State and Objects Databases and Files
+
+ioBroker is storing three different type of data:
+* **Objects** contain the meta data, descriptions and configuration values for all objects and states stored by ioBroker. Objects are created initially and sometimes updated, but more or less not changed that often
+* **Files** are all JSON-, Image- and other files that are accessible for all ioBroker adapters
+* **States** contain the real data from sensors, devices and objects which are updated often
+
+#### ioBroker In-Memory Database with JSON-File Storage
+
+By default the js-controller process is offering In-Memory-DBs at ports 9000 (for states) and 9001 (for objects/file). All adapter processes and other js-controller processes connect to these databases to read and store data.
+
+The In-Memory-DBs for States and Objects use JSON files to store the data. The files are stored after changes every 30 seconds and are backed up automatically. The files are stored in iobroker-data directory, the backups in a sub folder.
+
+The In-Memory-DBs work well for up to 10.000 Objects and normal State update frequencies. As soon as you have more Objects and States or you have states that are updated very often it is a better idea to use Redis as Database engine. You can detect this if js-controller process is using more and more CPU and/or the system feels slow.
+
+js-controller 1.x was using socket.io as communication protocol between the adapters and the In-Memory-DBs. Starting with js-controller 2.0 the communication protocol was changed to be a lightweight Redis protocol. 
+
+For the objects and states databases special additional logging of the redis protocl messages can be activated in iobroker.json
+
+```
+"objects": {
+  ...
+  "enhancedLogging": false
+}
+``` 
 
-To install REDIS on linux/debian just write: ```apt-get install redis-server``` .
 
-If you plan to use mulithost installation you must allow connections to redis from any address (default only 127.0.0.1).
+#### Redis as Database
+
+Redis is a well known industry In-Memory database which is optimized for speed and stability.
+
+##### Install Redis
+- Linux from here: https://redis.io/download
+- Windows from here: [https://github.com/MSOpenTech/redis/releases](https://github.com/MSOpenTech/redis/releases)
+
+e.g. for Linux:
```

Comment:
Either link to a manual or provide one yourself. Not both.

Reply:
kk